Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m

```diff
@@ -111,6 +214,73 @@ + (instancetype)providerWithAuth:(FIRAuth *)auth {
 }
 
 #pragma mark - Internal Methods
+/** @fn isVerifyAppURL:
+    @brief Parses a URL into all available query items.
+    @param URL The url to be checked against the authType string.
+    @return Whether or not the URL matches authType.
+ */
+- (BOOL)isVerifyAppURL:(nullable NSURL *)URL {
+  if (!URL) {
+    return NO;
+  }
+  NSURLComponents *actualURLComponents = [NSURLComponents new];
+  actualURLComponents.scheme = URL.scheme;
+  actualURLComponents.host = URL.host;
+  actualURLComponents.path = URL.path;
+  actualURLComponents.user = URL.user;
+  actualURLComponents.password = URL.password;
+
+  NSURLComponents *expectedURLComponents = [NSURLComponents new];
+  NSArray *strings = [_auth.app.options.clientID componentsSeparatedByString:@"."];
+  expectedURLComponents.scheme =
+      [[strings reverseObjectEnumerator].allObjects componentsJoinedByString:@"."];
+  expectedURLComponents.host = kFirebaseAuthURLHost;
```

Comment:
I feel using the string literal inline here is actually more readable than using a constant, just like `@"deep_link_id"` below. Same for next line. But I don't feel strongly so it's up to you.

Reply:
Agreed, these only make sense in this function for now.